osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -303,15 +303,12 @@ public WorkingBeatmap GetWorkingBeatmap(BeatmapInfo? importedBeatmap)
         {
             // Detached sets don't come with files.
             // If we seem to be missing files, now is a good time to re-fetch.
-            if (importedBeatmap?.BeatmapSet?.Files.Count == 0)
+            if (importedBeatmap?.IsManaged == false && importedBeatmap.BeatmapSet?.Files.Count == 0)
             {
-                using (var realm = contextFactory.CreateContext())
-                {
-                    var refetch = realm.Find<BeatmapInfo>(importedBeatmap.ID)?.Detach();
+                var refetch = contextFactory.Context.Find<BeatmapInfo>(importedBeatmap.ID)?.Detach();
```

Comment:
`GetWorkingBeatmap()` is routinely called outside of the update thread, so this change can't work in general (as evidenced by CI).

Reply:
Let's pretend that didn't happen.